mmdet3d/core/bbox/coders/km3d_decoder.py

```diff
@@ -0,0 +1,380 @@
+    ])
+
+    matrix_B[:, :, :] = data_B
+
+    data_C = torch.FloatTensor([
+        -length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        -length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y - width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y,
+        length * 0.5 * sin_rot_y + width * 0.5 * cos_rot_y
+    ])
+
+    matrix_C[:, :, :] = data_C
+
+    matrix_B = matrix_B - kp_norm * matrix_C  # 1 boxes 16
+
+    matrix_A_transposed = matrix_A.permute(0, 1, 3, 2)  # 1 boxes 3 16
+    # matrix_A_transposed in shape(boxes 3 16)
+    matrix_A_transposed = matrix_A_transposed.view(batch * boxes, 3, 16)
+    matrix_A = matrix_A.view(batch * boxes, 16, 3)  # boxes 16 3
+    matrix_B = matrix_B.view(batch * boxes, 16, 1).float()  # boxes 16 1
+
+    pinv = torch.bmm(matrix_A_transposed, matrix_A)  # boxes 3 3
+    pinv = torch.inverse(pinv)  # boxes 3 3
+
+    pinv = torch.bmm(pinv, matrix_A_transposed)  # boxes 3 16
+    pinv = torch.bmm(pinv, matrix_B)  # boxes 3 1
+    pinv = pinv.view(batch, boxes, 3, 1).squeeze(3)  # 1 boxes 3
+
+    return pinv, rot_y, kps
+
+
+def object_pose_decode(heat,
+                       wh,
+                       kps,
+                       dim,
+                       rot,
+                       prob=None,
+                       reg=None,
+                       hm_hp=None,
+                       hp_offset=None,
+                       K=100,
+                       meta=None,
+                       const=None):
+    """The entry for Decoding object's pose with final bboxes, scores, key
+    points, dimensions, heatmap score, yaw angle, position, probability, and
+    classes.
+
+    Args:
+        heat (torch.Tensor): Predictions from model in shape (batch_size *
+            channels * height * width)
+        wh (torch.Tensor): Widths and heights of bboxs
+        kps (torch.Tensor): Key points in shape (1 * batch_size * 18)
+        dim (int): dimensions of height, width, length in shape (1 * batch_size
+            * 3)
+        rot (torch.Tensor): Rotation values in shape (1* batch_size * 8)
+        prob (torch.Tensor): Probabilities
+        reg (torch.Tensor): Reg parameters
+        hm_hp (torch.Tensor): Heatmap hp parameters
```

Comment:
Don't use abbreviated "hp", at least rewrite the description, and it would be better to rename the variables